v2/dtos/requests/subscription.go

```diff
@@ -0,0 +1,110 @@
+//
+// Copyright (C) 2020-2021 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package requests
+
+import (
+	"encoding/json"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/errors"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/v2"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/v2/dtos"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/v2/dtos/common"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/v2/models"
+)
+
+// AddSubscriptionRequest defines the Request Content for POST Subscription DTO.
+// This object and its properties correspond to the AddSubscriptionRequest object in the APIv2 specification:
+// https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/support-notifications/2.x#/AddSubscriptionRequest
+type AddSubscriptionRequest struct {
```

Comment:
Please add factory method that takes parameters for the required properties on the Subscription DTO. Factory method should auto set properties like `ApiVersion`, `RequestId`, etc.
See this PR for example:
https://github.com/edgexfoundry/go-mod-core-contracts/pull/468